### ISSUE TITLE
`Layout/SpaceBeforeBrackets` sees spaces where they don't exist (?)

### ISSUE THREAD

**Issue by stdedos** [rubocop/rubocop#13356]
File to review: https://github.com/hashie/hashie/blob/d07b2a3a9fcb22e11a7af0fb19b436fa673dd5fe/lib/hashie/dash.rb#L99

## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation.

_idk_

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

```ruby
dash.rb:99:43: C: [Corrected] Layout/SpaceBeforeBrackets: Remove the space before the opening brackets.
      define_method(setter) { |value| self.[]=(property_name, value) }
                                          ^^^^
dash.rb:99:43: F: Lint/Syntax: unexpected token tLPAREN2
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
      define_method(setter) { |value| self(property_name, value) }
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.67.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, analyzing as Ruby 3.3, running on ruby 3.3.5) [x86_64-linux]
  - rubocop-performance 1.22.1
  - rubocop-rspec 3.1.0
```
